### PR TITLE
Full cleanup

### PR DESCRIPTION
**Pull request by harshilsharma63**

### CODE REVIEW

**Comment by harshilsharma63** [comment]
@ffdixon I've fixed the majority of the issues reported in full code review. For bigger changes, I've filed some issues which can be addressed later. This also allows us to prioritize the search reporting query optimization changes.

**Comment by cadavre** [comment]
Just made a quick review of PR. Looks to fix few of issues I was about to create.

Looking forward to get it merged!

**Comment by cadavre** [comment]
Please keep in mind that rename `salt` -> `secret` will likely broke backward compatibility because of plugin config stored with `salt` key.

Bet would be to provide migration or fallback.

**Comment by ffdixon** [comment]
A fallback makes sense so nothing current breaks.

File: server/bigbluebuttonapiwrapper/api/APIcalls.go [hunk 1]
```diff
@@ -19,32 +19,44 @@ package api
...
 //url of the BigBlueButton server
 var BaseUrl string
 
 //Secret of the BigBlueButton server
-var SALT string
+var salt string
 
-//Sets the BaseUrl and SALT
-func SetAPI(url string, salt string) {
+//Sets the BaseUrl and salt
```
[the remark this answers is not in the document]
Comment (harshilsharma63) [reply server/bigbluebuttonapiwrapper/api/APIcalls.go]: Sure, will refactor that.

**Comment by cadavre** [comment]
I'd suggest to release 2.1.0 with with being merged.

**Comment by harshilsharma63** [comment]
> Please keep in mind that rename `salt` -> `secret` will likely broke backward compatibility because of plugin config stored with `salt` key.
> 
> Best would be to provide migration or fallback.

Good catch! We can keep the setting key as original and just update the variable and display names.

**Comment by harshilsharma63** [comment]
Tested this on latest Mattermost v5.18, everything works great! Proceeding with merging.

**Comment by harshilsharma63** [comment]
@ffdixon can you merge this PR for me? I don't have permission to merge to `master`.

**Pull request merged**